File: R/Regression/Polynomial_Regression/polynomial_regression.r
dataset = read.csv(("Position_salaries.csv"))
dataset = dataset[2:3]


#Fitting Polynomial Regression to the dataset Set
dataset$Level2 = dataset$Level^2
dataset$Level3 = dataset$Level^3
dataset$Level4 = dataset$Level^4
poly_reg = lm(formula = Salary ~ .,data = dataset)

#visualizing the dataset
ggplot() + 
  geom_point(aes(x = dataset$Level, y = dataset$Salary), colour = 'red') + 
  geom_line(aes(x = dataset$Level, y = predict(poly_reg, newdata = dataset)), colour = 'blue') +
  ggtitle('R Truth or Bluff (Polynomial Regression)') +
  xlab('Level') +
  ylab('Salary')

y_pred = predict(poly_reg, data.frame(Level = 6.5, Level2 = 6.5^2, Level3 = 6.5^3, Level4 = 6.5^4))

File: Python/Regression/Polynomial_Regression/polynomial_regression.py
# -*- coding: utf-8 -*-
"""
Created on Mon Nov 11 19:06:16 2019

@author: PWL
"""
# Importing the libraries
import pandas as pd
import matplotlib.pyplot as plt
import numpy as np

#Importing the dataset
dataset = pd.read_csv('Position_Salaries.csv')
X = dataset.iloc[:, 1:2].values
y = dataset.iloc[:, 2].values


#Fitting Polynomial Regression into dataset
from sklearn.preprocessing import PolynomialFeatures
poly_reg = PolynomialFeatures(degree = 4)
X_poly = poly_reg.fit_transform(X)
poly_reg.fit(X_poly, y)
lin_reg_2 = LinearRegression()
lin_reg_2.fit(X_poly, y)


#Visualizing the Polynomial Regression result
X_grid = np.arange(min(X), max(X), 0.1)
X_grid = X_grid.reshape(len(X_grid), 1)
plt.scatter(X, y, color = 'red')
plt.plot(X_grid, lin_reg_2.predict(poly_reg.fit_transform(X_grid)), color='blue')
plt.title('Truth or Bluff (Polynomial Regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show() 

#Predicting new result with Polynomial Regression 
lin_reg_2.predict(poly_reg.fit_transform(6.5))